File: backend/internal/dto/investment.go
package dto

type InvestmentBreakdownDto struct {
	AssetType       string     `json:"asset_type"` // e.g. Reksadana, Saham, Deposito, Logam Mulia
	Amount          float64    `json:"amount"`
	FormattedAmount string     `json:"formatted_amount"`
	Percentage      float64    `json:"percentage"`
}

type InvestmentSummaryResponse struct {
	TotalInvestment MoneyValue               `json:"total_investment"`
	LiquidCash      MoneyValue               `json:"liquid_cash"`
	LiquidRatio     float64                  `json:"liquid_ratio"`
	InvestedRatio   float64                  `json:"invested_ratio"`
	Breakdown       []InvestmentBreakdownDto `json:"breakdown"`
	Trend           []MonthlyTrendDto        `json:"trend"`
}

type MonthlyTrendDto struct {
	Month           string     `json:"month"` // YYYY-MM
	Value           float64    `json:"value"`
	FormattedValue  string     `json:"formatted_value"`
}

File: frontend/src/services/investment.ts
import api from '../utils/api';
import type { MoneyValue } from './dashboard';

export interface InvestmentBreakdown {
  asset_type: string;
  amount: number;
  formatted_amount: string;
  percentage: number;
}

export interface MonthlyTrend {
  month: string;
  value: number;
  formatted_value: string;
}

export interface InvestmentSummaryResponse {
  total_investment: MoneyValue;
  liquid_cash: MoneyValue;
  liquid_ratio: number;
  invested_ratio: number;
  breakdown: InvestmentBreakdown[];
  trend: MonthlyTrend[];
}

export const investmentService = {
  async getInvestmentSummary(): Promise<InvestmentSummaryResponse> {
    const res = await api.get('/investment/summary');
    return res.data.data;
  },
};
export default investmentService;
